shell/platform/windows/window.cc

```diff
@@ -200,10 +201,36 @@ LRESULT Window::OnGetObject(UINT const message,
   }
 
   gfx::NativeViewAccessible root_view = GetNativeViewAccessible();
+  // TODO(schectman): UIA is currently disabled by default.
+  // https://github.com/flutter/flutter/issues/114547
+#ifdef FLUTTER_ENGINE_USE_UIA
   if (is_uia_request && root_view) {
+    if (!ax_fragment_root_) {
+      if (!ax_fragment_delegate_) {
+        ax_fragment_delegate_ =
+            std::make_unique<WindowAXFragmentRootDelegate>(*this);
+      }
+      ax_fragment_root_ = std::make_unique<ui::AXFragmentRootWin>(
+          window_handle_, ax_fragment_delegate_.get());
+    }
+
     // TODO(cbracken): https://github.com/flutter/flutter/issues/94782
     // Implement when we adopt UIA support.
+    // Retrieve UIA object for the root view.
+    Microsoft::WRL::ComPtr<IRawElementProviderSimple> root;
+    if (SUCCEEDED(ax_fragment_root_->GetNativeViewAccessible()->QueryInterface(
+            IID_PPV_ARGS(&root)))) {
```

Comment:
I see that Chrome doesn't have this `SUCCEEDED` check. Just curious, why did you add this? Feel free to keep as is, just asking for knowledge :)

Reply:
I do not know for sure why they do not use the check, but `QueryInterface` can return failure

Reply:
Can we not guarantee that `QueryInterface` will succeed here since we entirely own the logic?

Reply:
I don't know how accurate it is to say we entirely own the logic. ATL provides a default `QueryInterface` which calls `InternalQueryInterface`, which we own, which then calls, `CComObjectRootBase::InternalQueryInterface`, which is also a provided ATL implementation.